umh-core/pkg/fsm/redpanda/reconcile.go

```diff
@@ -220,16 +220,40 @@ func (r *RedpandaInstance) reconcileOperationalStates(ctx context.Context, servi
 		metrics.ObserveReconcileTime(metrics.ComponentRedpandaInstance, r.baseFSMInstance.GetID()+".reconcileOperationalStates", time.Since(start))
 	}()
 
+	r.baseFSMInstance.GetLogger().Infof("reconciling transition to %s from state %s", desiredState, currentState)
+
 	switch desiredState {
 	case OperationalStateActive:
 		return r.reconcileTransitionToActive(ctx, services, currentState, currentTime)
 	case OperationalStateStopped:
 		return r.reconcileTransitionToStopped(ctx, services, currentState)
+	case OperationalStateRestarting:
+		return r.reconcileTransitionToRestarting(ctx, services, currentState)
 	default:
 		return fmt.Errorf("invalid desired state: %s", desiredState), false
 	}
 }
 
+// reconcileTransitionToRestarting handles transitions when the desired state is Restarting.
+func (r *RedpandaInstance) reconcileTransitionToRestarting(ctx context.Context, services serviceregistry.Provider, currentState string) (err error, reconciled bool) {
```

Comment:
How about instead of introducing a restarting state, simply set it once to "stopping" by sending the stop event, then it should restart automatically (should finish the stopping, and should then start again)

Reply:
because restarting is not a desired state, it is just an action.

or alternatively, just work with an event, that can be send in running states only:
- reconcileTransitionToActive (and if the current state is then to_be_restarted, then call the restart function, if it is restarting, check whether the restart was completed)

mostly the logic remains the same, just that we don't have a desired state of restarting